### ISSUE TITLE
malformed json, when docliteid contains '\'

### ISSUE THREAD

**Issue by theonemcdonald** [dwgebler/doclite#11]
This might not be a bug:

If the document id contains `\`, the document won't save. I'm using a custom entity object class and using `collection->save($obj)` to save the object.

```
In DatabaseConnection.php line 219:

  Error executing query


In DatabaseConnection.php line 214:

  SQLSTATE[HY000]: General error: 1 malformed JSON
```

**Comment by dwgebler** [comment]
@theonemcdonald it might be a bug, I am not able to reproduce with a trivial example:

```
$db = new FileDatabase('../blog1/live/var/data.db', false, true, 1);
$users = $db->collection("users");
$user = $users->get();
$user->setId('an\id');
$user->save();
```
This is saved in the database as JSON:

```
{"__id":"an\\id"}
```

Can you provide a code sample which is triggering the error for you? Thanks.

**Comment by dwgebler** [comment]
@theonemcdonald okay yeah I can see where the bug happens when a document is mapped to a custom class where the internal ID field doesn't get properly escaped to JSON (i.e. changing the `\` to `\\`)). Should be easy to sort out, I'll try to get this fixed in the next couple of days and do a bugfix release.

**Comment by dwgebler** [comment]
@theonemcdonald this should be fixed in release 1.1.3, please upgrade and let me know if you have any further issues. Thanks.

**Comment by theonemcdonald** [comment]
Thanks! I will give this a shot this weekend, appreciate the quick work
